### PR TITLE
Drop offline mode

### PR DESCRIPTION
**Pull request by superm1**
One thing I want to call out that I chose to do is rather than re-assign all flags to rename the old flags to `*_UNUSED`.  As we're doing the ABI bump with 2.0.0 maybe we want to just slide them all up though, so feedback welcome on this.

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [x] Feature
- [ ] Documentation

### CODE REVIEW

File: plugins/genesys/fu-genesys-usbhub-device.c [hunk 1]
```diff
@@ -1878,11 +1878,9 @@ fu_genesys_usbhub_device_prepare(FuDevice *device,
...
 
 	fu_progress_set_id(progress, G_STRLOC);
-	fu_progress_add_step(progress, FWUPD_STATUS_SCHEDULING, 5, "detach");
-	for (guint i = 0; i < fw_types->len; i++) {
-		fu_progress_add_step(progress, FWUPD_STATUS_SCHEDULING, 100, NULL);
-		fu_progress_add_step(progress, FWUPD_STATUS_SCHEDULING, 100, NULL);
-	}
+	fu_progress_add_step(progress, FWUPD_STATUS_LOADING, 5, "detach");
+	for (guint i = 0; i < fw_types->len; i++)
+		fu_progress_add_step(progress, FWUPD_STATUS_LOADING, 100, NULL);
```
Comment (hughsie) [inline plugins/genesys/fu-genesys-usbhub-device.c:1883]: Not convinced about this one -- the original code looks correct to me...

**Comment by hughsie** [comment]
I kinda flip flop on removing all the offline stuff; on one hand it's ununsed and complex code, on the other hand it might get us out of a hole in the future. I think it might be wise to keep it around a bit longer.

File: plugins/genesys/fu-genesys-usbhub-device.c (hunk 1 above)
Comment (superm1) [reply plugins/genesys/fu-genesys-usbhub-device.c:1883]: Not with our definition of scheduling. It's meant to say it's running offline.

**Comment by superm1** [comment]
> I kinda flip flop on removing all the offline stuff; on one hand it's ununsed and complex code, on the other hand it might get us out of a hole in the future. I think it might be wise to keep it around a bit longer.

how about if we change it from meson tri-state feature to a boolean and set it disabled a while?

**Comment by hughsie** [comment]
> and set it disabled a while

I guess that's pragmatic.

**Comment by superm1** [comment]
Alternative in https://github.com/fwupd/fwupd/pull/6953

**Pull request closed without merging**